curated_transformers/cli/quantize.py

```diff
@@ -0,0 +1,97 @@
+from io import BytesIO
+from pathlib import Path
+import spacy
+from spacy import Language
+from spacy.cli import app
+from thinc.api import Model, PyTorchShim
+import torch
+from torch.nn import Embedding, Module
+from torch.quantization import qconfig
+from typer import Argument as Arg, Option
+
+from ..models.torchscript_wrapper import to_torchscript_wrapper
+from ..pipe import Transformer
+
+
+@app.command("quantize-transformer")
+def quantize_cli(
+    model_path: Path = Arg(..., help="Model to quantize", exists=True, allow_dash=True),
+    output_path: Path = Arg(..., help="Output directory to store quantized model in"),
+    skip_embeds: bool = Option(
+        False, "--skip-embeds", help="Do not quantize embeddings"
+    ),
+    skip_linear: bool = Option(
+        False, "--skip-linear", help="Do not quantize linear layers"
+    ),
+):
+    """
+    Quantize a curated-transformers model.
+    """
+    nlp = spacy.load(model_path)
+    nlp_quantize_dynamic(nlp, skip_embeds=skip_embeds, skip_linear=skip_linear)
+    nlp.to_disk(output_path)
+
+
+def size_of_model(model):
+    filelike = BytesIO()
+    torch.save(model.state_dict(), filelike)
+    return filelike.tell()
+
+
+def nlp_quantize_dynamic(
+    nlp: Language, *, skip_embeds: bool = False, skip_linear: bool = False
+):
+    for pipe_name, pipe in nlp.components:
+        # We only quantize curated transformers, other models may not
+        # be able to deal with quantization and/or TorchScript.
+        if not isinstance(pipe, Transformer):
+            continue
+
+        model = pipe.model.get_ref("transformer")
+
+        if model.name != "pytorch":  # Probably already quantized.
```

Comment:
We could be explicit here and compare it to `pytorch_script` instead.

Reply:
👍 much better